### PR TITLE
bugfix for local refs url support for SPA and recursively use the describedby editor

### PR DESCRIPTION
**Pull request by hybtalented**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks backward-compatibility?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | 
| Updated README/docs?   | ✔️
| Added CHANGELOG entry?   | ❌

1. the local refs for `describedby` editor may wook poorly if i change `window.location.hash` ( which is common for single page application) or `window.locaiton.query` after editor instance is created. 
2. the `$extend` function didn't copy a array appropriately(deeply), which leads to a problem for recursively use of `describedby` editor. for example the shema likes:
``` json

缩进量：  引号 显示控制 
格式化JSON：
{
  "title": "Property for Sale", 
  "$ref": "#/definitions/single", 
  "definitions": {
    "multiple": {
      "type": "array", 
      "items": {
        "$ref": "#/definitions/single"
      }
    }, 
    "single": {
      "id": "basic", 
      "type": "object", 
      "properties": {
        "propertyName": {
          "type": "string", 
          "title": "Property Name", 
          "description": "The name of the property being sold", 
          "minLength": 5
        }, 
        "propertyType": {
          "type": "string", 
          "title": "Property Type", 
          "enum": [
            "house", 
            "car", 
            "yacht", 
            "multiple"
          ]
        }, 
        "propertyTypeOptions": {
          "type": [
            "object", 
            "array"
          ], 
          "title": "Property Options", 
          "watch": {
            "tag": "basic.propertyType"
          }, 
          "links": [
            {
              "rel": "describedBy", 
              "href": "#/definitions/{{tag}}"
            }
          ]
        }
      }, 
      "required": [
        "propertyName", 
        "propertyType"
      ]
    }, 
    "house": {
      "properties": {
        "address": {
          "type": "string", 
          "title": "Address"
        }, 
        "rooms": {
          "type": "integer", 
          "title": "Number Of Rooms"
        }
      }
    }, 
    "car": {
      "properties": {
        "brand": {
          "type": "string", 
          "title": "Brand", 
          "enum": [
            "Audi", 
            "BMW", 
            "Chevrolet", 
            "Opel", 
            "Mitsubishi"
          ]
        }, 
        "color": {
          "type": "string", 
          "title": "Color", 
          "format": "color"
        }, 
        "year": {
          "type": "string", 
          "title": "Build Year"
        }
      }
    }, 
    "yacht": {
      "properties": {
        "name": {
          "type": "string", 
          "title": "Name"
        }, 
        "model": {
          "type": "string", 
          "title": "Model"
        }, 
        "year": {
          "type": "integer", 
          "title": "Year"
        }, 
        "yachtLength": {
          "type": "integer", 
          "title": "Length"
        }
      }
    }
  }
}
```

### CODE REVIEW

File: src/editors/describedby.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ export var DescribedByEditor = AbstractEditor.extend({
       return '';
     }); */
 
-    this.schema.links.splice(0, 1)
+    this.schema.links.splice(i, 1)
```
Comment (hybtalented) [inline src/editors/describedby.js]: delete the correct link

**Comment by pmk65** [comment]
@hybtalented Does this also fix #443 ?

**Comment by hybtalented** [comment]
> @hybtalented Does this also fix #443 ?

I think so. The may be cause by the same reason, that is `$extend` didn't copy the `Array` in schema appropriately,  when the decribedby editor created，`this.schema.links.splice(i, 1)`  will delete the describedby links from the origin schema.

**Comment by pmk65** [comment]
@schmunk42 For some reason the Selenium tests fail in Firefox when using the custom helper `pressKey`. This used to work before, so something must have changed.

**Comment by hybtalented** [comment]
> For some reason the Selenium tests fail in Firefox (but not Chrome) when using the custom helper `pressKey`. This used to work before, so something must have changed.

@pmk65,  it looks like the `CodeceptJS `package is upgraded. The last pull request which  Selenium tests passed, has the `CodeceptJS` of  `v2.3.2`, however CodeceptJS package is upgrading to `v2.3.3` recently.

**Comment by hybtalented** [comment]
> `CodeceptJS`

https://github.com/Codeception/CodeceptJS/blob/2.3.3/CHANGELOG.md may do some help。

**Comment by schmunk42** [comment]
Thank you!

**Pull request merged**